store/tikv/2pc.go

```diff
@@ -1523,17 +1523,39 @@ func (c *twoPhaseCommitter) tryAmendTxn(ctx context.Context, startInfoSchema Sch
 		c.doingAmend = true
 		defer func() { c.doingAmend = false }()
 		if keysNeedToLock.len() > 0 {
-			pessimisticLockBo := NewBackofferWithVars(ctx, pessimisticLockMaxBackoff, c.txn.vars)
 			lCtx := &kv.LockCtx{
-				ForUpdateTS:  c.forUpdateTS,
-				LockWaitTime: kv.LockNoWait,
+				Killed:        c.lockCtx.Killed,
+				ForUpdateTS:   c.forUpdateTS,
+				LockWaitTime:  c.lockCtx.LockWaitTime,
+				WaitStartTime: time.Now(),
 			}
-			err = c.pessimisticLockMutations(pessimisticLockBo, lCtx, keysNeedToLock)
-			if err != nil {
-				logutil.Logger(ctx).Warn("amend pessimistic lock has failed", zap.Error(err), zap.Uint64("txnStartTS", c.startTS))
-				return false, err
+			tryTimes := uint(0)
+			retryLimit := config.GetGlobalConfig().PessimisticTxn.MaxRetryCount
+			for tryTimes < retryLimit {
+				pessimisticLockBo := NewBackofferWithVars(ctx, pessimisticLockMaxBackoff, c.txn.vars)
+				err = c.pessimisticLockMutations(pessimisticLockBo, lCtx, keysNeedToLock)
+				if err != nil {
+					// KeysNeedToLock won't change, so don't async rollback pessimistic locks here for write conflict.
+					if terror.ErrorEqual(kv.ErrWriteConflict, err) {
+						newForUpdateTSVer, err := c.store.CurrentVersion()
+						if err != nil {
+							return false, errors.Trace(err)
+						}
+						lCtx.ForUpdateTS = newForUpdateTSVer.Ver
+						c.forUpdateTS = newForUpdateTSVer.Ver
+						logutil.Logger(ctx).Info("amend pessimistic lock pessimistic retry lock",
```

Comment:
Will this log be noisy?

Reply:
I think it'ok to leave more logs for this new feature currently and it won't happen frequently.

Reply:
OK, but I think it will eventually be complained about.